File: src/algorithms/count-subarrays.js
/**
 * Given an integer array nums, return the number of subarrays of length 3 such
 * that the sum of the first and third numbers equals exactly half of the
 * second number.
 *
 * @format
 * @param {number[]} nums - An array of numbers.
 * @returns {number} - Returns a count of conditions that are met.
 */

function countSubarrays(nums) {
  count = 0;
  for (let i = 0; i < nums.length - 2; i++) {
    if ((nums[i] + nums[i + 2]) * 2 === nums[i + 1]) {
      count += 1;
    }
  }
  return count;
}

File: src/algorithms/count-subarrays.py
def countSubarrays(nums):
    """Given an integer array nums, return the number of subarrays of length 3 
    such that the sum of the first and third numbers equals exactly half of the 
    second number.

    Args:
        nums (list[int]): A list of integers.

    Returns:
        int: Returns a count of subarrays that meet condition.
    """
    count = 0
    for i in range(len(nums) - 2):
        subarraySum = nums[i] + nums[i + 2]
        if nums[i + 1] / 2 == subarraySum:
            count += 1
    return count
